x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -445,11 +536,11 @@ export class DataRecognizer {
   }
 
   async loadIndexPatterns() {
-    return await this.savedObjectsClient.find({ type: 'index-pattern', perPage: 1000 });
+    return await this.savedObjectsClient!.find({ type: 'index-pattern', perPage: 1000 });
```

Comment:
It actually might be worth checking that `this.savedObjectsClient` exists here instead of using non-null assertion here. Even though right now `loadIndexPatterns` is only used after `savedObjectsClient` has been set this check would guard against any future updates/additions to this class that might try to use this method differently.

Reply:
I believe it should be passed to the constructor, not the method. In that case, no check is required. I just wanted to avoid any code changes in this PR, but probably it worth a small refactoring.

Reply:
Refactored in https://github.com/elastic/kibana/pull/57206/commits/63ea75b825f7659b09b62f62ab420f33651637ed